File: applicatie/src/main/java/com/vdzon/administratie/mongo/TestDataGenerator.java
package com.vdzon.administratie.mongo;

import com.vdzon.administratie.model.*;
import org.mongodb.morphia.Datastore;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

public class TestDataGenerator {

    public static void buildTestData(String username, String name, String passwd, boolean admin, Datastore datastore) {
        System.out.println("Maak admin user");
        List<Contact> adresboek = new ArrayList<>();
        List<Bestelling> bestellingen = new ArrayList<>();
        List<Factuur> facturen = new ArrayList<>();
        List<Rekening> rekeningen = new ArrayList<>();
        List<Afschrift> afschriften = new ArrayList<>();
        List<Declaratie> declaraties = new ArrayList<>();
        Administratie administratie = Administratie.newBuilder().uuid(getNewUuid()).bestellingen(bestellingen).facturen(facturen).adresboek(adresboek).rekeningen(rekeningen).afschriften(afschriften)
                .declaraties(declaraties).administratieGegevens(AdministratieGegevens.newBuilder().build()).build();
        List<Administratie> administraties = new ArrayList<>();
        administraties.add(administratie);
        Gebruiker gebruiker = Gebruiker.newBuilder().uuid(getNewUuid()).name(name).username(username).password(passwd).admin(admin).administraties(administraties).build();
        datastore.save(gebruiker);
    }

    private static String getNewUuid() {
        return UUID.randomUUID().toString();
    }


}

File: storage/impl/src/main/java/com/vdzon/administratie/mongo/TestDataGenerator.kt
package com.vdzon.administratie.mongo

import com.mongodb.client.MongoDatabase
import com.vdzon.administratie.model.*
import com.vdzon.administratie.model.boekingen.Boeking
import org.litote.kmongo.getCollection
//import org.mongodb.morphia.Datastore

import java.time.LocalDate
import java.util.ArrayList
import java.util.Random
import java.util.UUID

object TestDataGenerator {

    fun buildTestData(username: String, name: String, passwd: String, admin: Boolean, datastore: MongoDatabase) {
        println("Maak admin user")
        val adresboek = ArrayList<Contact>()
        val bestellingen = ArrayList<Bestelling>()
        val facturen = ArrayList<Factuur>()
        val rekeningen = ArrayList<Rekening>()
        val afschriften = ArrayList<Afschrift>()
        val declaraties = ArrayList<Declaratie>()
        val boekingen = ArrayList<Boeking>()
        val administratie = Administratie(
                newUuid,
                AdministratieGegevens(),
                bestellingen,
                facturen,
                adresboek,
                rekeningen,
                afschriften,
                declaraties,
                boekingen)
        val administraties = ArrayList<Administratie>()
        administraties.add(administratie)
        val gebruiker = Gebruiker(newUuid, name, username, passwd, admin, administraties)
        datastore.getCollection<Gebruiker>().insertOne(gebruiker)
    }

    private val newUuid: String
        get() = UUID.randomUUID().toString()


}
